src/date-range-input/date-range-input.tsx

```diff
@@ -275,6 +283,31 @@ export const DateRangeInput = ({
         }
     };
 
+    const handleFixedRangeDateChange = (val: string) => {
+        if (isDateUnselectable(val)) {
+            isUnselectable.current = true;
+            return;
+        }
+
+        actions.changeStart(val);
+        calendarRef.current.setCalendarDate(val);
+        isUnselectable.current = false;
+
+        if (!val) {
+            if (!withButton) {
+                actions.resetRange({ start: "", end: "" });
+                onChange?.("", "");
+            }
+            return;
+        }
```

Comment:
I think we can skip the `withButton` check here, so that the end date is cleared if there's no start date, to match with the calendar state
```suggestion
        if (!val) {
            actions.resetRange({ start: "", end: "" });
            onChange?.("", "");
            return;
        }
```